test/concurrent/ThreadPoolTest.ooc

```diff
@@ -27,6 +27,11 @@ ThreadPoolTest: class extends Fixture {
 			expect(promise3 wait())
 			expect(promise4 wait(), is false)
 			expect(promise5 wait(), is true)
+			expect(promise1 _state as Int, is equal to(_PromiseState Finished as Int))
+			expect(promise2 _state as Int, is equal to(_PromiseState Finished as Int))
+			expect(promise3 _state as Int, is equal to(_PromiseState Finished as Int))
+			expect(promise4 _state as Int, is equal to(_PromiseState Cancelled as Int))
+			expect(promise5 _state as Int, is equal to(_PromiseState Finished as Int))
```

Comment:
There should be some public (overrideable) method that returns the "appropriate" state. For the promises/futures returned by threadpool, the appropriate state is not this state but rather the one in their respective task.

This leads me to believe it may be incorrect that `_Synchronizer` has a state, and we should move it to `_ThreadPromise` and `_ThreadFuture<T>` classes. In these classes, `getState()` will return this state, and in the threadpool versions, `getState()` will return the state in their tasks.